constantine/ethereum_bls_signatures.nim

```diff
@@ -510,3 +513,62 @@ func batch_verify*[Msg](pubkeys: openArray[PublicKey], messages: openarray[Msg],
   if verified:
     return cttEthBls_Success
   return cttEthBls_VerificationFailure
+
+func alloc_batch_sig_accumulator*(): ptr BatchSigAccumulator {.libPrefix: prefix_ffi.} =
+  ## Allocates using `alloc0` for a `BatchSigAccumulator`
+  ## so that it can remain an incomplete struct in the C header.
+  result = cast[ptr BatchSigAccumulator](alloc0(sizeof(BatchSigAccumulator)))
```

Comment:
Use `allocHeapAligned`. https://github.com/mratsim/constantine/blob/285aeb280f84b889195529bf821e2801423bcb08/constantine/platforms/allocs.nim#L101-L107

We don't want to use Nim allocator in Constantine for low-level primitives so that:
- We can easily debug / track allocations with regular tools like valgrind
- Memory is deallocated and not retained for caching
- We don't bring the Nim runtime into other languages, this means we don't need to call `NimMain` before using constantine.dll / constantine.so and it's a nice UX improvement.
- Allocator can be swapped, Nim's is tuned for real-time, but Constantine is used in 24/7 applications and in that case might benefit from jemalloc for the batch accumulator.